lib/core/pktmbuf/pktmbuf.h

```diff
@@ -1153,6 +1194,101 @@ const char *cne_get_tx_ol_flag_name(uint64_t mask);
  */
 int cne_get_tx_ol_flag_list(uint64_t mask, char *buf, size_t buflen);
 
+/**
+ * Return the metadata buffer address
+ *
+ * @param m
+ *   The pktmbuf_t structure pointer
+ * @return
+ *   NULL on error or pointer to metadata buffer
+ */
+static inline void *
+pktmbuf_metadata(const pktmbuf_t *m)
+{
+    pktmbuf_info_t *p;
+
+    if (!m)
+        return NULL;
+
+    if (((p = m->pooldata) != NULL) && p->metadata)
+        return CNE_PTR_ADD(p->metadata, (m->meta_index * p->metadata_bufsz));
+    else
+        return CNE_PTR_ADD(m, sizeof(pktmbuf_t)); /* default to metadata in pktmbuf headroom */
+}
+
+/**
+ * Return the size of the metadata buffer if external or headroom size if internal buffer.
+ *
+ * @param m
+ *   The pktmbuf_t structure pointer
+ * @return
+ *   -1 if pktmbuf pointer is NULL or size of metadata buffer or headroom size if internal.
+ */
+static inline uint32_t
```

Comment:
Should you returned a int32_t if you plan on returning -1? YOu could create a `define PKTMBUF_INV_MD_SZ (u32)-1` and return that instead if you want ot keep it a uint

Reply:
Use int32_t